### PR TITLE
CLOUDP-79621: Get all Cloud Provider Access Roles

### PR DESCRIPTION
**Pull request by andreaangiolillo**
## Proposed changes

_Jira ticket:_ CLOUDP-79621

## Checklist

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run `make fmt` and formatted my code

## Further comments

```
./bin/mongocli atlas cloudProvider accessRole  list
ID                         PROVIDER   ATLAS AWS ACCOUNT ARN            UNIQUE EXTERNAL ID
5ffc85265582a85ac3a71aae   AWS        arn:aws:iam::299602853325:root   0cffcad4-d258-4569-bf1e-2662a37ce230
5ffc854d35237e28cf76e2d7   AWS        arn:aws:iam::299602853325:root   9b51f6fa-d8ee-44ba-a564-8fef746738fc
5ffc85555582a85ac3a71afd   AWS        arn:aws:iam::299602853325:root   c5f3e624-de82-4a0e-bf96-22ca4ff1188c
5ffc855a5582a85ac3a71b05   AWS        arn:aws:iam::299602853325:root   4645114e-4c60-47e2-aef4-fd0cc617d8f6
5ffc90b602f8822ef207a622   AWS        arn:aws:iam::299602853325:root   5675c1d2-e403-4b11-9683-531e09ddb2f5
5ffc90ee02f8822ef207a680   AWS        arn:aws:iam::299602853325:root   e4f1238f-9781-4abe-bcab-7eb2e0ae0508
5ffd60e815a4f913021b470f   AWS        arn:aws:iam::299602853325:root   09aedddd-353c-4b25-9851-c39bdef35ab8
```
```
➜  ~/workspace/mongoCLI git:(CLOUDP-79621) ✗ ./bin/mongocli atlas cloudProvider accessRole  list -o json
{
  "awsIamRoles": [
    {
      "atlasAWSAccountArn": "arn:aws:iam::299602853325:root",
      "atlasAssumedRoleExternalId": "0cffcad4-d258-4569-bf1e-2662a37ce230",
      "createdDate": "2021-01-11T17:04:38Z",
      "providerName": "AWS",
      "roleId": "5ffc85265582a85ac3a71aae"
    },
    {
      "atlasAWSAccountArn": "arn:aws:iam::299602853325:root",
      "atlasAssumedRoleExternalId": "9b51f6fa-d8ee-44ba-a564-8fef746738fc",
      "createdDate": "2021-01-11T17:05:17Z",
      "providerName": "AWS",
      "roleId": "5ffc854d35237e28cf76e2d7"
    },
    {
      "atlasAWSAccountArn": "arn:aws:iam::299602853325:root",
      "atlasAssumedRoleExternalId": "c5f3e624-de82-4a0e-bf96-22ca4ff1188c",
      "createdDate": "2021-01-11T17:05:25Z",
      "providerName": "AWS",
      "roleId": "5ffc85555582a85ac3a71afd"
    },
    {
      "atlasAWSAccountArn": "arn:aws:iam::299602853325:root",
      "atlasAssumedRoleExternalId": "4645114e-4c60-47e2-aef4-fd0cc617d8f6",
      "createdDate": "2021-01-11T17:05:30Z",
      "providerName": "AWS",
      "roleId": "5ffc855a5582a85ac3a71b05"
    },
    {
      "atlasAWSAccountArn": "arn:aws:iam::299602853325:root",
      "atlasAssumedRoleExternalId": "5675c1d2-e403-4b11-9683-531e09ddb2f5",
      "createdDate": "2021-01-11T17:53:58Z",
      "providerName": "AWS",
      "roleId": "5ffc90b602f8822ef207a622"
    },
    {
      "atlasAWSAccountArn": "arn:aws:iam::299602853325:root",
      "atlasAssumedRoleExternalId": "e4f1238f-9781-4abe-bcab-7eb2e0ae0508",
      "createdDate": "2021-01-11T17:54:54Z",
      "providerName": "AWS",
      "roleId": "5ffc90ee02f8822ef207a680"
    },
    {
      "atlasAWSAccountArn": "arn:aws:iam::299602853325:root",
      "atlasAssumedRoleExternalId": "09aedddd-353c-4b25-9851-c39bdef35ab8",
      "createdDate": "2021-01-12T08:42:16Z",
      "providerName": "AWS",
      "roleId": "5ffd60e815a4f913021b470f"
    }
  ]
}
```

### CODE REVIEW

File: internal/cli/atlas/cloudproviders/accessroles/description.go [hunk 1]
```diff
@@ -14,5 +14,6 @@
 package accessroles
 
 const (
-	accessRole = "Manage AWS IAM role access in Atlas."
+	accessRole = "Manage IAM role access in Atlas."
+	list       = "List IAM role access in Atlas."
```
Comment (gssbzn) [inline internal/cli/atlas/cloudproviders/accessroles/description.go:18]:
```suggestion
	accessRole = "Manage AWS IAM role access in Atlas."
	list       = "List AWS IAM role access in Atlas."
```

File: internal/cli/atlas/cloudproviders/accessroles/list.go [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+}
+
+func (opts *ListOpts) initStore() error {
+	var err error
+	opts.store, err = store.New(config.Default())
+	return err
+}
+
+var listTemplate = `ID	PROVIDER{{range .AWSIAMRoles}}
+{{.RoleID}}	{{.ProviderName}}{{end}}
```
Comment (gssbzn) [inline internal/cli/atlas/cloudproviders/accessroles/list.go:40]: [nit] provider is always AWS, can we please add ARN and external ID as well here?

**Review by gssbzn — CHANGES_REQUESTED** [review]
two small nits

File: internal/cli/atlas/cloudproviders/accessroles/list.go (hunk 2 above)
Comment (andreaangiolillo) [reply internal/cli/atlas/cloudproviders/accessroles/list.go:40]: I thought that the reason why this command is not inside the AWS directory is that, in the future, the list command would show different providers as well. Am I wrong? If I am wrong, why aren’t we moving this command inside the aws folder

Comment (gssbzn) [reply internal/cli/atlas/cloudproviders/accessroles/list.go:40]: sorry, my comment is not about removing the provider but that only listing ID and provider name is not helpful, as they are all AWS, users may need more info like ARNs to know the difference

**Review by gssbzn — CHANGES_REQUESTED** [review]
Can you please an e2e tests for the list command, this should work we can wait until we have delete for the create

**Comment by andreaangiolillo** [comment]
there isn’t a delete command

**Review by gssbzn — APPROVED** [review]
Lovely 🎉

**Pull request merged**